homeassistant/components/onvif/parsers.py

```diff
@@ -12,15 +14,15 @@
 PARSERS: Registry[str, Callable[[str, Any], Coroutine[Any, Any, Event]]] = Registry()
 
 
-def datetime_or_zero(value: str) -> datetime:
-    """Convert strings to datetimes, if invalid, return datetime.min."""
+def datetime_or_zero(value: str) -> datetime | None:
+    """Convert strings to datetimes, if invalid, return None."""
     # To handle cameras that return times like '0000-00-00T00:00:00Z' (e.g. hikvision)
     try:
         ret = dt_util.parse_datetime(value)
```

Comment:
I think this would be easier:

```suggestion
        return dt_util.parse_datetime(value)
```

Reply:
This whole method van be removed, the place `datetime_or_zero` is used, can be replaced by just `dt_util.parse_datetime(value)` directly (as that already returns `None` and doesn't raise `ValueError` at all.

Reply:
> This whole method van be removed, the place `datetime_or_zero` is used, can be replaced by just `dt_util.parse_datetime(value)` directly (as that already returns `None` and doesn't raise `ValueError` at all.

But this was the the problem that #66357 was solving. A ValueError was generated on a date produced by a hikvision camera.

Am I missing something?

Reply:
@frenck `parse_datetime` does raise:
```
Python 3.9.9 (main, Jan 13 2022, 12:36:30)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from homeassistant.util import dt as dt_util
>>> dt_util.parse_datetime('0000-00-00T00:00:00Z')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/erik/development/home-assistant_fork/homeassistant/util/dt.py", line 206, in parse_datetime
    return dt.datetime(**kws)
ValueError: year 0 is out of range
```